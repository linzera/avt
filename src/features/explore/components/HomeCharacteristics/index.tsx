import React from 'react';
import {View, StyleSheet, Platform} from 'react-native';
import Typography from '~/components/Typography';

import BathIcon from '@explore/assets/svg/bath.svg';
import PoolIcon from '@explore/assets/svg/pool.svg';
import RoomsIcon from '@explore/assets/svg/rooms.svg';
import OccupancyIcon from '@explore/assets/svg/user.svg';

const styles = StyleSheet.create({
  details: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginBottom: 12,
    marginTop: 8,
    paddingHorizontal: 8,
    paddingBottom: Platform.select({ios: 16}),
    flexWrap: 'wrap',
  },
  item: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export const CHARACTERISTICS_MAPPING = {
  pool: {
    icon: <PoolIcon />,
    label: 'Pool',
  },
  bathrooms: {
    icon: <BathIcon />,
    label: 'Bathrooms',
  },
  occupancy: {
    icon: <OccupancyIcon />,
    label: '',
  },
  rooms: {
    icon: <RoomsIcon />,
    label: 'Bedrooms',
  },
};

interface Props {
  children?: React.ReactElement<ItemProps>[];
}

interface ItemProps {
  showLabel?: boolean;
  item: keyof typeof CHARACTERISTICS_MAPPING;
  count?: number;
}

const Item = ({showLabel, item, count}: ItemProps) => {
  const {icon, label} = CHARACTERISTICS_MAPPING[item];

  return (
    <View style={styles.item}>
      {icon}
      <Typography component="text12">
        <If condition={!!count}>{count}</If>
        <If condition={!!showLabel}> {label}</If>
      </Typography>
    </View>
  );
};

function HomeCharacteristics(props: Props) {
  return <View style={styles.details}>{props.children}</View>;
}

HomeCharacteristics.Item = Item;

export default HomeCharacteristics;
