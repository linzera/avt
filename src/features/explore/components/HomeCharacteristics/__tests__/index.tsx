import React from 'react';
import {render} from '@testing-library/react-native';
import HomeCharacteristics, {CHARACTERISTICS_MAPPING} from '..';

describe('<HomeCharacteristics />', () => {
  it('should render properly', () => {
    const {toJSON} = render(<HomeCharacteristics />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render all characteristics properly', () => {
    const {toJSON} = render(
      <HomeCharacteristics>
        {Object.keys(CHARACTERISTICS_MAPPING).map(item => (
          <HomeCharacteristics.Item
            key={item}
            item={item as keyof typeof CHARACTERISTICS_MAPPING}
            count={1}
            showLabel
          />
        ))}
      </HomeCharacteristics>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  describe('<HomeCharacteristics.Item />', () => {
    it('should render properly', () => {
      const {queryByText} = render(
        <HomeCharacteristics.Item item="rooms" count={1} showLabel />,
      );

      expect(queryByText(/1/)).not.toBeNull();
      expect(queryByText(/Bedrooms/)).not.toBeNull();
    });

    it('should not render count if has not count or it is 0', () => {
      const {queryByText} = render(
        <HomeCharacteristics.Item item="rooms" count={0} showLabel />,
      );

      expect(queryByText(/0/)).toBeNull();
      expect(queryByText(/Bedrooms/)).not.toBeNull();
    });

    it('should not render label given false showLabel prop', () => {
      const {queryByText} = render(
        <HomeCharacteristics.Item item="rooms" count={1} showLabel={false} />,
      );

      expect(queryByText(/1/)).not.toBeNull();
      expect(queryByText(/Bedrooms/)).toBeNull();
    });
  });
});
