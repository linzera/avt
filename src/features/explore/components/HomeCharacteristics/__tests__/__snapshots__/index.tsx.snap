// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeCharacteristics /> should render all characteristics properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-evenly",
      "marginBottom": 12,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingHorizontal": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.80)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
        }
      }
    >
      1
       
      Pool
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.80)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
        }
      }
    >
      1
       
      Bathrooms
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.80)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
        }
      }
    >
      1
       
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <SvgMock />
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.80)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
        }
      }
    >
      1
       
      Bedrooms
    </Text>
  </View>
</View>
`;

exports[`<HomeCharacteristics /> should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-evenly",
      "marginBottom": 12,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingHorizontal": 8,
    }
  }
/>
`;
